fabric-dimensions-v1/src/main/java/net/fabricmc/fabric/mixin/dimension/MixinMinecraftServer.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.dimension;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import net.minecraft.world.gen.chunk.ChunkGenerator;
+import net.minecraft.world.gen.chunk.DebugChunkGenerator;
+import net.minecraft.world.gen.chunk.FlatChunkGenerator;
+import net.minecraft.world.gen.chunk.SurfaceChunkGenerator;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.SimpleRegistry;
+import net.minecraft.world.World;
+import net.minecraft.world.dimension.DimensionOptions;
+import net.minecraft.world.gen.GeneratorOptions;
+
+@Mixin(MinecraftServer.class)
+public class MixinMinecraftServer {
+	// Replace the ChunkGenerator with the Current Seed and Skip Vanilla Dimensions
+	@Redirect(
+			method = "createWorlds",
+			at = @At(
+					value = "INVOKE",
+					target = "Lnet/minecraft/world/gen/GeneratorOptions;getDimensionMap()Lnet/minecraft/util/registry/SimpleRegistry;"
+			)
+	)
+	private SimpleRegistry<DimensionOptions> onGetDimensionMap(GeneratorOptions generatorOptions) {
+		SimpleRegistry<DimensionOptions> dimensionMap = generatorOptions.getDimensionMap();
+
+		dimensionMap.getIds().forEach(dimensionId -> {
+			if (isVanillaDimension(dimensionId)) {
+				return;
+			}
+
+			DimensionOptions dimensionOptions = dimensionMap.get(dimensionId);
+
+			if (isVanillaChunkGenerator(dimensionOptions.chunkGenerator)) {
+				return;
+			}
+
+			dimensionOptions.chunkGenerator = dimensionOptions.chunkGenerator.withSeed(generatorOptions.getSeed());
+		});
+
+		return dimensionMap;
+	}
+
+	@Unique
+	private static boolean isVanillaDimension(Identifier identifier) {
+		return identifier.equals(World.OVERWORLD.getValue())
+				|| identifier.equals(World.NETHER.getValue())
+				|| identifier.equals(World.END.getValue());
+	}
+
+	@Unique
+	private static boolean isVanillaChunkGenerator(ChunkGenerator chunkGenerator) {
+		return chunkGenerator instanceof SurfaceChunkGenerator
+				|| chunkGenerator instanceof FlatChunkGenerator
+				|| chunkGenerator instanceof DebugChunkGenerator;
+	}
```

Comment:
Will this not work if people use these generators in their dimensions?

Reply:
I do this to skip datapack added dimensions because vanilla datapack dimension can only use vanilla ChunkGenerator. Normal dimension mods create unique dimensions and do not use vanilla generators (If a mod really wants to use vanilla ChunkGenerator, it can create a new ChunkGenerator and wrap a vanilla generator https://github.com/qouteall/FabricDimensionTest1.16/blob/master/src/main/java/net/fabricmc/example/ExampleChunkGenerator.java). (And I don't know how to know whether a dimension is being added by datapack)

Reply:
I forgot that vanilla chunk generators can be extended. Going to change that to getClass() == ...

Reply:
Humm, yes thats better.

I see some other possible options:
1. An interface you put on your ChunkGenerator, not ideal as it wont work if you are using the vanilla generators
2. A registry of some sort that allows you to specify what dims use the server's seed
3. A way to define in the json that the chunk generator should use the server's seed. (I dont think this is currenly feasible).

This does seem like something mojang will need to work on soon. Would be nice to get some other people's input on this as I know some others have been looking into it.